magento/magento2
# Avoid duplicate loading of configuration files

### Description (*)

While I was investigating why some pages in back-office are extremely slow on my local environment, I found out that we are currently loading multiple times all `layouts.xml` files in the `\Magento\Theme\Model\PageLayout\Config\Builder` class.

Through this change, I propose to only load them one time by adding a new property to store the result and reuse it for the next calls. There is no structural change and the **loading time is reduced by almost 40%** on my local environment, as you can see with the screenshots from Blackfire below.

**Previously:** 
![Blackfire screenshot](https://duaw26jehqd4r.cloudfront.net/items/040x3c2r3U3V2C1r2n2U/getConfigFiles-before.png)

**Now:**
![Blackfire screenshot](https://duaw26jehqd4r.cloudfront.net/items/0Z0N1l1f1U0R0c3s0g1j/%5B21ac1bc61b4fca85535fd1301ddd76d0%5D_getConfigFiles-after.png)

**Comparison:**
![Blackfire screenshot](https://duaw26jehqd4r.cloudfront.net/items/3I0H2Y0Y013V032k3u2P/%5Bb764c16e271860806c6def46f5b542cc%5D_getConfigFiles-comparison.png)



### Manual testing scenarios (*)

1. Log in to the back office.
2. Go to Content > Pages.
3. Look at the loading time... 😄 

Please note that I'm using Docker for Mac [with this environment](https://github.com/EmakinaFR/docker-magento2), and the application is in `developer` mode with all caches activated. The latest version is used for both Docker and Magento.

### Contribution checklist (*)
